test/integration/fake_upstream.h

```diff
@@ -446,10 +446,17 @@ class FakeHttpConnection : public Http::ServerConnectionCallbacks, public FakeCo
 
     // Network::ReadFilter
     Network::FilterStatus onData(Buffer::Instance& data, bool) override {
+      Http::Status status;
       try {
-        parent_.codec_->dispatch(data);
+        status = parent_.codec_->dispatch(data);
+        // TODO(#10878): Remove this when exception removal is complete. It is currently in
+        // migration. Soon we won't need to catch these exceptions, as they'll be propagated through
+        // the error statuses callbacks and returned from dispatch.
       } catch (const Http::CodecProtocolException& e) {
-        ENVOY_LOG(debug, "FakeUpstream dispatch error: {}", e.what());
+        status = Http::codecProtocolError(e.what());
+      }
```

Comment:
This looks similar to the factorizable stanza above, though there's one less 'catch' in it. Re-use the suggested factorered helper-method out here anyway?

Reply:
done